spec/statsd/sender_spec.rb

```diff
@@ -62,6 +62,22 @@
       end
     end
 
+    context 'when a ThreadError is raised starting the sender thread' do
+      let(:thread_class) do
+        class_double(Thread)
+      end
+
+      before do
+        allow(thread_class).to receive(:new).and_raise(ThreadError, "ThreadError")
```

Comment:
Minor:

```suggestion
        expect(thread_class).to receive(:new).and_raise(ThreadError, "ThreadError")
```

Using `expect` here adds a tiny extra level of checks against the test passing trivially, e.g. if the method does not do anything. This is useful in tests that check "it doesn't break" as otherwise they can pass for the wrong reasons.

Alternatively, you can expect on the `logger` receiving a message as a way of "observing" that the correct path was exercised in the test.